File: rtmp/src/main/java/net/ossrs/rtmp/BitrateManager.java
package net.ossrs.rtmp;

/**
 * Created by pedro on 10/07/19.
 *
 * Calculate video and audio bitrate per second
 */
public class BitrateManager {

  private long bitrate;
  private long timeStamp = System.currentTimeMillis();
  private ConnectCheckerRtmp connectCheckerRtmp;

  public BitrateManager(ConnectCheckerRtmp connectCheckerRtsp) {
    this.connectCheckerRtmp = connectCheckerRtsp;
  }

  public synchronized void calculateBitrate(long size) {
    bitrate += size;
    long timeDiff = System.currentTimeMillis() - timeStamp;
    if (timeDiff >= 1000) {
      connectCheckerRtmp.onNewBitrateRtmp((int) (bitrate / (timeDiff / 1000f)));
      timeStamp = System.currentTimeMillis();
      bitrate = 0;
    }
  }
}

File: rtmp/src/main/java/com/pedro/rtmp/utils/BitrateManager.kt
package com.pedro.rtmp.utils

/**
 * Created by pedro on 8/04/21.
 *
 * Calculate video and audio bitrate per second
 */
open class BitrateManager(private val connectCheckerRtmp: ConnectCheckerRtmp) {

  private var bitrate: Long = 0
  private var timeStamp = System.currentTimeMillis()

  @Synchronized
  fun calculateBitrate(size: Long) {
    bitrate += size
    val timeDiff = System.currentTimeMillis() - timeStamp
    if (timeDiff >= 1000) {
      connectCheckerRtmp.onNewBitrateRtmp((bitrate / (timeDiff / 1000f)).toLong())
      timeStamp = System.currentTimeMillis()
      bitrate = 0
    }
  }
}